//
// Copyright 2021 The Feast Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//


syntax = "proto3";
package feast.core;

option go_package = "github.com/feast-dev/feast/go/protos/feast/core";
option java_outer_classname = "RequestFeatureViewProto";
option java_package = "feast.proto.core";

import "feast/core/DataSource.proto";

message RequestFeatureView {
    // User-specified specifications of this feature view.
    RequestFeatureViewSpec spec = 1;
}

// Next available id: 7
message RequestFeatureViewSpec {
    // Name of the feature view. Must be unique. Not updated.
    string name = 1;

    // Name of Feast project that this feature view belongs to.
    string project = 2;

    // Request data which contains the underlying data schema and list of associated features
    DataSource request_data_source = 3;

    // Description of the request feature view.
    string description = 4;

    // User defined metadata.
    map<string,string> tags = 5;

    // Owner of the request feature view.
    string owner = 6;
}
